cmd/nginx/main.go

```diff
@@ -170,16 +170,23 @@ func handleSigterm(ngx *controller.NGINXController, exit exiter) {
 	signal.Notify(signalChan, syscall.SIGTERM)
 	<-signalChan
 	klog.Info("Received SIGTERM, shutting down")
+	// 1. TODO(elvinefendi) at this point we should immediately fail readiness probe and sleep for some configurable time
```

Comment:
For this we should add a check in https://github.com/kubernetes/ingress-nginx/blob/17e788b8e11536e037e52cfd56e5849e7bb5405f/internal/ingress/controller/checker.go#L38 checking for `n.isShuttingDown`